gitopsaddon/test-cycle.sh

```
@@ -787,6 +932,428 @@ verify_gitopsaddon_labels() {
+wait_for_agent_pod_stable() {
+    local cluster=$1
+    local max_attempts=${2:-12}
+    local last_phase="" last_ready="" last_restarts=""
+    local stable=false
+    for attempt in $(seq 1 "$max_attempts"); do
+        local phase ready restarts
+        phase=$(on_cluster "$cluster" get pod -n "$ARGOCD_NS" -l app.kubernetes.io/name=acm-openshift-gitops-agent-agent -o jsonpath='{.items[0].status.phase}' 2>/dev/null || echo "")
+        ready=$(on_cluster "$cluster" get pod -n "$ARGOCD_NS" -l app.kubernetes.io/name=acm-openshift-gitops-agent-agent -o jsonpath='{.items[0].status.containerStatuses[0].ready}' 2>/dev/null || echo "")
+        restarts=$(on_cluster "$cluster" get pod -n "$ARGOCD_NS" -l app.kubernetes.io/name=acm-openshift-gitops-agent-agent -o jsonpath='{.items[0].status.containerStatuses[0].restartCount}' 2>/dev/null || echo "")
+        if [ "$phase" = "Running" ] && [ "$ready" = "true" ] && \
+           [ "$phase" = "$last_phase" ] && [ "$ready" = "$last_ready" ] && [ "$restarts" = "$last_restarts" ]; then
```

Shellcheck (0.11.0)

[warning] Expanding an array without an index only gives the first element.

(SC2128)